code/modules/spells/spell_code.dm

```diff
@@ -99,11 +99,7 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 	return
 
 /spell/proc/is_valid_target(var/target, mob/user)
-	if(!(spell_flags & INCLUDEUSER) && target == usr)
-		return 0
-	if(get_dist(usr, target) > range) //Shouldn't be necessary but a good check in case of overrides
-		return 0
-	return istype(target, /mob/living)
+	return (target in view_or_range(range, user, selection_type))
```

Comment:
The primary problem with this is now its going to call view or range for every possible thing in the targets list, which could be rather intensive based on what spells are being cast.